rpc/amqp-rpc.go

```diff
@@ -407,7 +411,31 @@ func (rpc *AmqpRPCServer) Start(c cmd.Config) error {
 			select {
 			case msg, ok := <-msgs:
 				if ok {
-					rpc.processMessage(msg)
+					if rpc.maxGoroutines > 0 && atomic.LoadInt64(&rpc.currentGoroutines) >= rpc.maxGoroutines {
+						response := RPCResponse{Error: wrapError(core.InternalServerError("RPC server has spawned too many Goroutines"))}
```

Comment:
This isn't really an InternalServerError. That's an HTTP notion, but these shouldn't be exposed all the way back to HTTP. Define a new error type so that (at some point) RPC clients can respond nicely to it. Also, pre-marshal it at startup so we can reply without having to marshal.